realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -672,6 +702,20 @@ Builder schema(Class<? extends RealmModel> firstClass, Class<? extends RealmMode
          * @return the created {@link RealmConfiguration}.
          */
         public RealmConfiguration build() {
+            // Check that readOnly() was applied to legal configuration. Right now it should only be allowed if
+            // an assetFile is configured
+            if (readOnly) {
+                if (initialDataTransaction != null) {
+                    throw new IllegalStateException("This Realm is marked as read-only. Read-only Realms cannot use initialData(Realm.Transaction).");
+                }
+                if (assetFilePath == null) {
+                    throw new IllegalStateException("Only Realms provided using 'assetFile(path)' can be marked read-only. No such Realm was provided.");
+                }
+                if (deleteRealmIfMigrationNeeded) {
+                    throw new IllegalStateException("'deleteRealmIfMigrationNeeded()' and read-only Realms cannot be combined");
```

Comment:
@zaki50 This change will be a bit bigger than I would like since our current code path only assume you can copy the file if it doesn't exist already. I would prefer creating a separate issue for this we can track?